mbedtls/src/cipher/raw/serde.rs

```diff
@@ -117,8 +71,75 @@ impl<Op: Operation> Serialize for Cipher<Op, Traditional, CipherData> {
     }
 }
 
-impl<'de, Op: Operation> Deserialize<'de> for Cipher<Op, Traditional, CipherData> {
-    fn deserialize<D>(d: D) -> Result<Cipher<Op, Traditional, CipherData>, D::Error>
+unsafe fn serialize_raw_cipher<S: Serializer>(mut cipher_context: cipher_context_t)
+    -> Result<SavedRawCipher, S::Error> {
```

Comment:
Return `Result<SavedRawCipher, &'static str>` instead so you don't have to pass `S` around.